signatureverifier/src/main/kotlin/io/github/rakutentech/signatureverifier/verification/AesEncryptor.kt

```diff
@@ -0,0 +1,156 @@
+package io.github.rakutentech.signatureverifier.verification
+
+import android.os.Build
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import android.util.Base64
+import android.util.Log
+import androidx.annotation.RequiresApi
+import androidx.annotation.VisibleForTesting
+import com.google.gson.Gson
+import com.google.gson.JsonParseException
+import io.github.rakutentech.signatureverifier.SignatureVerifier
+import java.security.KeyStore
+import java.security.KeyStoreException
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.GCMParameterSpec
+
+@SuppressWarnings("TooGenericExceptionCaught")
+internal class AesEncryptor @VisibleForTesting constructor(
```

Comment:
I can see there are a lot of exceptions here. Is that solution more stable than EncryptedSharedPref? I mean is there a changes that this code won't work on some device?

Reply:
Yes, this is more stable. The use of `Keystore` and `Cipher` has less risk to be an issue in devices.
The catching of exception is just a precaution to handle all possible OS exception (defensive) and not allow SDK to crash.
This way, if the encryption fails, SDK will still function correctly (always fetch from backend) and not store the key in cache.